middleware/middleware.go

```diff
@@ -15,14 +16,29 @@ import (
 
 type tokenContextKey struct{}
 
-const (
-	defaultSessionName = "oidc-middleware"
+var baseLogAttr = slog.String("component", "oidc-middleware")
+
+func errAttr(err error) slog.Attr { return slog.String("err", err.Error()) }
+
+// SessionData contains the data this middleware needs to save/restore across
+// requests.
+type SessionData struct {
+	// State for an in-progress auth flow.
+	State string `json:"oidc_state"`
+	// ReturnTo is where we should navigate to at the end of the flow
+	ReturnTo string `json:"oidc_return_to"`
+	// IDToken is the id_token for the current logged in user
+	IDToken string `json:"oidc_id_token"`
+	// RefreshToken is the refresh token for this OIDC session. It is only
+	// persisted if a secure session store is used.
+	RefreshToken string `json:"oidc_refresh_token"`
+}
 
-	sessionKeyOIDCState        = "oidc-state"
-	sessionKeyOIDCReturnTo     = "oidc-return-to"
-	sessionKeyOIDCIDToken      = "oidc-id-token"
-	sessionKeyOIDCRefreshToken = "oidc-refresh-token"
-)
+// Sessions are used for managing state across requests.
+type Sessions interface {
+	GetSession(*http.Request) (*SessionData, error)
+	SaveSession(http.ResponseWriter, *http.Request, *SessionData) error
+}
```

Comment:
nit, but WDYT of `type SessionStoreV2` with `Get` and `Put`? Less stutter-y. More generally I wonder if it's even worth maintaining backward compatibility in lstoll/oidc?

Reply:
Would you change the value on the `Middleware` struct to `SessionStoreV2` as well?

Also I waffle a lot on the stutter thing on interfaces. Really generic stuff like `Get` often conflicts on implementations if you're passing something that does multiple things, I don't like having to have a sub-struct to work around that. But yeah, stutter.

Reply:
> Would you change the value on the Middleware struct to SessionStoreV2 as well?

Probably, yes.

> Really generic stuff like Get often conflicts on implementations 

I mean things like the Reader interface is also super generic, and isn't a problem in practice I dunno. https://pkg.go.dev/golang.org/x/crypto/acme/autocert#DirCache also.

Reply:
Fiiiiiine

Reply:
Dropped backwards compatibility, renamed things